cmd/bucket-encrypt.go

```diff
@@ -0,0 +1,167 @@
+/*
+ * MinIO Client (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Encrypt 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	json "github.com/minio/mc/pkg/colorjson"
+	"github.com/minio/mc/pkg/probe"
+
+	"github.com/minio/minio/pkg/console"
+)
+
+var bucketEncryptCmd = cli.Command{
+	Name:   "encrypt",
+	Usage:  "manage bucket encryption config",
+	Action: mainBucketEncrypt,
+	Before: setGlobalsFromContext,
+	Flags:  globalFlags,
+	CustomHelpTemplate: `NAME:
+   {{.HelpName}} - {{.Usage}}
+ 
+ USAGE:
+   {{.HelpName}} TARGET [info | clear | [SSE-S3 | SSE-KMS kms-key]]
+ 
+ FLAGS:
+   {{range .VisibleFlags}}{{.}}
+   {{end}}
+ EXAMPLES:
+   1. Display bucket encryption status for bucket "mybucket".
+      {{.Prompt}} {{.HelpName}} myminio/mybucket info
+
+   2. Enable SSE-S3 auto encryption on bucket "mybucket" for alias "myminio".
+      {{.Prompt}} {{.HelpName}} myminio/mybucket sse-s3
+   
+   3. Enable SSE-KMS auto encryption with kms key on bucket "mybucket" for alias "s3".
+      {{.Prompt}} {{.HelpName}} myminio/mybucket sse-kms arn:aws:kms:us-east-1:xxx:key/xxx
+ 
+ 
+   3. Remove auto encryption config on bucket "mybucket" for alias "myminio".
+      {{.Prompt}} {{.HelpName}} myminio/mybucket clear 
+`,
+}
+
+// checkBucketEncryptSyntax - validate all the passed arguments
+func checkBucketEncryptSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) < 2 || len(ctx.Args()) > 3 {
+		cli.ShowCommandHelpAndExit(ctx, "encrypt", 1) // last argument is exit code
+	}
+}
+
+type bucketEncryptMessage struct {
+	Op         string `json:"op"`
+	Status     string `json:"status"`
+	URL        string `json:"url"`
+	Encryption struct {
+		Algorithm string `json:"algorithm"`
+		KeyID     string `json:"keyId,omitempty"`
+	} `json:"encryption"`
```

Comment:
```suggestion
	} `json:"encryption, omitempty"`
```

Reply:
don't think this should be omitted - if there is no encryption configuration the struct should be empty

Reply:
no, this is required for the clear command
```
$ mc bucket encrypt play/kannappan clear --json
{
 "op": "clear",
 "status": "success",
 "url": "play/kannappan",
 "encryption": {
  "algorithm": ""
 }
}
```